File: lib/rin/gui/Padding.cpp
#include "rin/gui/Padding.h"
namespace rin::gui
{
    Padding::operator Vec4<float>() const
    {
        return Vec4{left,top,right,bottom};
    }
    Padding::operator Vec2<>() const
    {
        return Vec2<>{left + right,top + bottom};
    }
    Padding::Padding(const float inAll)
    {
        left = inAll;
        top = inAll;
        right = inAll;
        bottom = inAll;
    }
    Padding::Padding(const float inHorizontal, const float inVertical)
    {
        left = inHorizontal;
        top = inVertical;
        right = inHorizontal;
        bottom = inVertical;
    }
    Padding::Padding(const float inLeft, const float inRight, const float inTop, const float inBottom)
    {
        left = inLeft;
        top = inTop;
        right = inRight;
        bottom = inBottom;
    }
}

File: Rin.Engine/Views/Padding.cs
using System.Numerics;

namespace Rin.Engine.Views;

public class Padding : ICopyable<Padding>
{
    public float Bottom;
    public float Left;
    public float Right;
    public float Top;

    public Padding()
    {
    }

    public Padding(float val)
    {
        Left = val;
        Right = val;
        Top = val;
        Bottom = val;
    }

    public Padding(float horizontal, float vertical)
    {
        Left = horizontal;
        Right = horizontal;
        Top = vertical;
        Bottom = vertical;
    }

    public Padding Copy()
    {
        return new Padding
        {
            Left = Left,
            Right = Right,
            Top = Top,
            Bottom = Bottom
        };
    }

    public static implicit operator Vector4(Padding p)
    {
        return new Vector4(p.Left, p.Top, p.Right, p.Bottom);
    }

    public static implicit operator Padding(float val)
    {
        return new Padding(val);
    }
}